Source/WebKit/UIProcess/mac/WebContextMenuProxyMac.mm

```diff
@@ -385,24 +385,12 @@ - (void)menuDidClose:(NSMenu *)menu
 #endif // ENABLE(IMAGE_ANALYSIS_ENHANCEMENTS)
 }
 
+#if !HAVE(SHARING_SERVICE_PICKER_STANDARD_SHARE_MENU_ITEM)
 static void getStandardShareMenuItem(NSArray *items, void (^completionHandler)(NSMenuItem *))
 {
-#if HAVE(NSSHARINGSERVICEPICKER_ASYNC_MENUS)
-    // FIXME (<rdar://problem/54551500>): Replace this with the async variant of +[NSMenuItem standardShareMenuItemForItems:] when it's available.
-    auto sharingServicePicker = adoptNS([[NSSharingServicePicker alloc] initWithItems:items]);
-    [sharingServicePicker setStyle:NSSharingServicePickerStyleMenu];
-    [sharingServicePicker getMenuWithCompletion:^(NSMenu *shareMenu) {
-        ASSERT(isMainThread());
-        shareMenu.delegate = (id <NSMenuDelegate>)sharingServicePicker.get();
-        auto shareMenuItem = adoptNS([[NSMenuItem alloc] initWithTitle:WEB_UI_STRING("Share", "Title for Share context menu item.") action:nil keyEquivalent:@""]);
-        [shareMenuItem setRepresentedObject:sharingServicePicker.get()];
-        [shareMenuItem setSubmenu:shareMenu];
-        completionHandler(shareMenuItem.get());
-    }];
-#else
```

Comment:
This isn't safe to remove. `HAVE_NSSHARINGSERVICEPICKER_ASYNC_MENUS` is true on macOS 12.

While this code might not run on macOS 13 due to the logic below, it still applies to macOS 12.